File: solution45.go
func jump(nums []int) int {
	end := 0
	rightmost := 0
	step := 0
	size := len(nums)

	for i := 0; i < size-1; i++ {
		if i <= rightmost {
			if i+nums[i] > rightmost {
				rightmost = i + nums[i]
			}
			if i == end {
				end = rightmost
				step++
			}
		}
	}

	return step
}

File: solution45.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int step = 0;
        int rightmost = 0;
        int end = 0;
        int size = nums.size();

        for (int i = 0; i < size - 1; i ++) {
            if (i <= rightmost) {
                rightmost = max(rightmost, i + nums[i]);
                if (i == end) {
                    end = rightmost;
                    step ++;
                }
            }
        }
        return step;
    }
};
